samples/algorithms/JointMeasurement.qs

```diff
@@ -0,0 +1,24 @@
+/// # Sample
+/// Joint Measurement
+///
+/// # Description
+/// Joint measurements, also known as Pauli measurements, are a generalization
+/// of 2-outcome measurements to multiple qubits and other bases.
+namespace Sample {
+    @EntryPoint()
+    operation Main () : Result {
+        // In Q#, the `Measure` operation performs a joint measurement of one or
+        // more qubits in the specified Pauli bases. It takes two arguments: an
+        // array of single-qubit Pauli values indicating the tensor product
+        // factors on each qubit, and a register of qubits to be measured.
+
+        // The below code uses a joint measurement as a way to check the parity
+        // of two qubits.
+        use qs = Qubit[2];
+        H(qs[0]);
+        CNOT(qs[0], qs[1]);
+        let result = Measure([PauliZ, PauliZ], qs);
```

Comment:
Might be useful to mention that Joint Measurements can be used to get information about a collection of qubits without necessarily collapsing the state of any one qubit.

Reply:
For example here, the result will always be even parity (I think that is `Zero`) and the state of the qubits are unchanged, as we would have already known the parity must be zero from the previous code. If you want to emphasize how the joint measurement can change the state of the system of qubits, you could show the joint measurement doing the entangling instead of the `CNOT`, performing an `X(qs[1])` if the result is odd parity (`One`), though that example uses a measurement result in a condition.

Reply:
I think using the joint measurement to perform an entanglement might be too advanced for a simple sample and is a topic that belongs in the Katas.

However, the example of getting information about a collection of qubits makes sense. We could modify the example to be something like this:
```qsharp
    operation Main() : (Result, Result[]) {
        // Prepare an entangled state.
        use qs = Qubit[2];  // |00〉
        H(qs[0]);           // 1/sqrt(2)(|00〉 + |10〉)
        CNOT(qs[0], qs[1]); // 1/sqrt(2)(|00〉 + |11〉)

        // Show the quantum state before performing the joint measurement.
        DumpMachine();

        // The below code uses a joint measurement as a way to check the parity
        // of the first two qubits. In this case, the parity measurement result
        // will always be `Zero`.
        // Notice how the state was not collapsed by the joint measurement.
        let parityResult = Measure([PauliZ, PauliZ], qs[...1]); 
        DumpMachine();

        // However, if we perform a measurement just on the first qubit, we can
        // see how the state collapses.
        let firstQubitResult = M(qs[0]);
        DumpMachine();

        // Measuring the last qubit does not change the state quantum state
        // since the state of the second qubit collapsed when the first qubit 
        // was measured because they were entangled.
        let secondQubitResult = M(qs[1]);
        DumpMachine();

        ResetAll(qs);
        return (parityResult, [firstQubitResult, secondQubitResult]);
    }
```